read-fonts/src/lib.rs

```diff
@@ -1,4 +1,51 @@
-//! Parsing OpentType tables.
+//! Reading OpenType tables
+//!
+//! This crate provides memory safe zero-allocation parsing of font files.
+//! It is unopinionated, and attempts to provide raw access to the underlying
+//! font data as it is described in the [OpenType specification][spec].
+//!
+//! This crate is intended for use by other parts of a font stack, such as a
+//! shaping engine or a glyph rasterizer.
+//!
+//! In addition to raw data access, this crate may also provide reference
+//! implementations of algorithms for interpreting that data, where such an
+//! implementation is required for the data to be useful. For instance, we
+//! provide functions for [mapping codepoints to glyph identifiers][cmap-impl]
+//! using the `cmap` table, or for [decoding entries in the `name` table][NameString].
+//!
+//! ## Codegen
+//!
+//! The majority of the code in this crate is auto-generated. For more information
+//! on our use of codegen, see the [codegen tour].
+//!
+//! # Related projects
+//!
+//! - [`write-fonts`] is a companion crate for creating/modifying font files
+//! - [`skrifa`] provides access to glyph outlines and metadata (essentially a
+//!   Rust alternative to [freetype])
+//!
+//! # Example
+//!
+//! ```no_run
+//! # let path_to_my_font_file = std::path::Path::new("");
+//! use read_fonts::{FontRef, TableProvider};
+//! let font_bytes = std::fs::read(path_to_my_font_file).unwrap();
+//! let font = FontRef::new(&font_bytes).expect("failed to read font data");
```

Comment:
`FontRef::from_index(&font_bytes, 0)` will work for both single fonts and collection files. Should we nudge users in that direction?

Reply:
I wanted to just keep this dead-simple, but I'll update the docs for `FontRef::new` to mention `from_index`, and add a comment inline. :)